async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -139,4 +148,80 @@ object AsyncHttpClient {
     Headers(headers.asScala.map { header =>
       Header(header.getKey, header.getValue)
     }.toList)
+
+  def webSocketResource[F[_]](config: AsyncHttpClientConfig = defaultConfig)(
+      implicit F: ConcurrentEffect[F]): Resource[F, WebSocketClient[F]] =
+    Resource
+      .make(F.delay(new DefaultAsyncHttpClient(config)))(c => F.delay(c.close()))
+      .map(client =>
+        WebSocketClient[F] { req =>
+          for {
+            receiveQueue <- Queue.unbounded[F, WebSocketFrame]
```

Comment:
This is not ideal but I don't see a way to apply backpressure short of blocking the listener.